lib/Activity.py

```diff
@@ -810,12 +811,13 @@ def monitor_session(self, workflow_id, stime):
 
     def get_workflow(self, workflow):
         try:
-            wf = self.config.connection.run(f"kubectl -n argo get Workflow/{workflow} -o yaml")
+            ret_code = self.nlsapi.get_workflow(workflow)
```

Comment:
If we're replacing 'kubectl' with this new call are we sure the new call is going to be resilient?    Should we have some logic that can retry the call in the case of an API failure?    Otherwise I'm not sure this will solve the problem.

Reply:
that's true, we are currently investigating more on the logic to see if retry needs to be included in the code (cli or nls)

Reply:
@don-bahls-hpe  we have added a retry logic in nls. you can review https://github.com/Cray-HPE/cray-nls/pull/218